### PR TITLE
Don't leak temp files on failures

### PR DESCRIPTION
**Pull request by rhatdan**
#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Which issue(s) this PR fixes:

Fixes: https://github.com/containers/podman/issues/22642

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@edsantiago @Luap99 @nalind @flouthoc @TomSweeneyRedHat @giuseppe @vrothberg PTAL

File: image.go [hunk 1]
```diff
@@ -952,7 +952,7 @@ func (i *containerImageSource) GetBlob(ctx context.Context, blob types.BlobInfo,
 // makeExtraImageContentDiff creates an archive file containing the contents of
 // files named in i.extraImageContent.  The footer that marks the end of the
 // archive may be omitted.
-func (i *containerImageRef) makeExtraImageContentDiff(includeFooter bool) (string, digest.Digest, int64, error) {
+func (i *containerImageRef) makeExtraImageContentDiff(includeFooter bool) (_ string, _ digest.Digest, _ int64, Err error) {
```
Comment (Luap99) [inline image.go:955]: please do not use upper case for a local var, I suggest a name like `retErr`

File: run_common.go [hunk 2]
```diff
@@ -1659,7 +1659,7 @@ func (b *Builder) getTmpfsMount(tokens []string, idMaps IDMaps) (*specs.Mount, e
 	return &volumes[0], nil
 }
 
-func (b *Builder) getSecretMount(tokens []string, secrets map[string]define.Secret, idMaps IDMaps, workdir string) (*specs.Mount, string, error) {
+func (b *Builder) getSecretMount(tokens []string, secrets map[string]define.Secret, idMaps IDMaps, workdir string) (_ *specs.Mount, _ string, Err error) {
```
Comment (Luap99) [inline run_common.go:1662]: same here

**Review by Luap99 — COMMENTED** [review]
LGTM, I haven't looked to closely at the source of the problem but these two cases seem logical.

I don't think it fixes the podman issue though as there seem to be leaks also only on the podman side.

**Comment by TomSweeneyRedHat** [comment]
`Err` to `retErr` then LGTM

**Review by flouthoc — APPROVED** [review]
LGTM

**Review by Luap99 — APPROVED** [review]
/lgtm

**Pull request merged**